src/init.c

```diff
@@ -201,7 +201,7 @@ void jl_task_frame_noreturn(jl_task_t *ct);
 
 JL_DLLEXPORT void jl_atexit_hook(int exitcode)
 {
-    if (jl_all_tls_states == NULL)
+    if (jl_atomic_load_relaxed(&jl_all_tls_states) == NULL)
         return;
 
     jl_task_t *ct = jl_current_task;
```

Comment:
```suggestion

```

Reply:
Need to use `jl_get_current_task` if it might be NULL

Reply:
```suggestion
    jl_task_t *ct = jl_get_current_task();
```

this had an assume-non-null on it, so you cannot use it

the `jl_task_frame_noreturn` call also needs a conditional guard on `ct` being defined

Reply:
Ok done!